src/Chapter4.hs

```diff
@@ -628,8 +666,10 @@ Can you implement a monad version of AND, polymorphic over any monad?
 
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
+
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM b1 b2 = b1 >>= (\b -> if not b then pure False else b2)
+-- why pure False and not Just False? :/
```

Comment:
Because `False :: Bool`, and this is not the thing that you want to get according to the type. Also, as you are already in Monad that means you can't get out of it and start working with pure values.